utils/data.cpp

```diff
@@ -0,0 +1,253 @@
+#include "data.hpp"
+#include <QDebug>
+
+namespace jbrush {
+
+// Constructor makes some simple assignments
+Data::Data(Canvas* theCanvas) : mainCanvas(theCanvas), currentDirectory(STARTING_DIRECTORY){}
+
+void Data::Load()
```

Comment:
This method is huge. Let's break it up. Some ideas:
 - Private function to get the next shape from the QTextStream (so the body of the loop is only one line)
 - Private functions to construct a text shape, line shape, or other shape.

Reply:
Jeez, I dunno why I didn't do that sooner! The code is a lot easier to work with now that I've taken this suggestion ;)